### DISCUSSION TITLE
Why does Next.js take up so much memory?

### DISCUSSION THREAD

**Discussion by Judge-Paul** [vercel/next.js#56279] [Help]
### Summary

I am currently using the latest version of Next.js (13.5.1) and after running the server for some time I noticed it starts using up to 2gb of memory.
Note: I am working on a small project with only 3 routes and about 15 total .js files in my app directory. 
Why is it consuming so much memory and is there anything I can do to optimize it or use less memory cause it's making my PC to slow down during development

**Comment by khuezy** [comment]
Are you still seeing memory issues on 14.x? It sounds like you have a memory leak somewhere in your code or a library you're using.
I'm running a standalone app (latest `next`) in a container w/ Nodejs 21 and it starts at ~70MB and once `next` runs, it idles at around ~120MB

**Reply by Judge-Paul** [reply]
I haven't used Next 14, I had these issues with Next 13 and I switched to pages router for my new projects
